src/main/java/org/broadinstitute/hellbender/tools/walkers/varianteval/AbstractPluginDescriptor.java

```diff
@@ -0,0 +1,102 @@
+package org.broadinstitute.hellbender.tools.walkers.varianteval;
+
+import org.broadinstitute.barclay.argparser.CommandLineException;
+import org.broadinstitute.barclay.argparser.CommandLinePluginDescriptor;
+
+import java.util.*;
+
+abstract public class AbstractPluginDescriptor<T> extends CommandLinePluginDescriptor {
+    protected final Map<String, T> allDiscoveredPlugins = new HashMap<>();
+
+    protected List<String> pluginPackageNames;
+    protected Class<T> pluginBaseClass;
+    protected Class<T> defaultPluginBaseClass;
```

Comment:
This is not really generic for an abstract plugin descriptor: it assumes that the defaults are always extending a concrete classs - this does not happen with, for example, `ReadFilter` (where defaults are tool-specific)

Reply:
Yes, I actually made this change locally as well.  Again, this class as-is isnt really for prime-time as a product-wide Abstact class.  I think we'd want to start by pulling the relevant parts out of ReadFilterPluginDescriptor.  My implementation also doesnt deal with the bits related to plugins supplying command-line arguments.